src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -138,8 +154,13 @@ private void WriteDepsFile(string depsFilePath)
             IEnumerable<ReferenceInfo> referenceAssemblyInfos =
                 ReferenceInfo.CreateReferenceInfos(ReferenceAssemblies);
 
+            // If there is a generated asset file. The projectContext will have project reference.
+            // So remove it from directReferences to avoid duplication
+            var projectReferenceExistedInProjectContext = projectContext != null;
```

Comment:
>projectReferenceExistedInProjectContext [](start = 16, length = 39)

Naming suggestion: `projectContextHasProjectReferences`

Elsewhere it was confusing me that this seemed to be referring to a single project refer rather than the general set of all project references.